cpp/src/gandiva/function_registry_common.h

```diff
@@ -74,31 +74,31 @@ typedef std::unordered_map<const FunctionSignature*, const NativeFunction*, KeyH
 // - NULL handling is of type NULL_IF_NULL
 //
 // The pre-compiled fn name includes the base name & input type names. eg. add_int32_int32
-#define BINARY_SYMMETRIC_SAFE_NULL_IF_NULL(NAME, TYPE)                             \
-  NativeFunction(#NAME, DataTypeVector{TYPE(), TYPE()}, TYPE(), kResultNullIfNull, \
-                 ARROW_STRINGIFY(NAME##_##TYPE##_##TYPE))
+#define BINARY_SYMMETRIC_SAFE_NULL_IF_NULL(NAME, ALIASES, TYPE)          \
+  NativeFunction(#NAME, ALIASES, DataTypeVector{TYPE(), TYPE()}, TYPE(), \
```

Comment:
Macros which take initializer_lists are tricky because the lack of a paren means that an initializer_list with more than one element is considered two tokens. For example:

```c++
#define VECTOR(T, elements) ::std::vector<T>(elements)
VECTOR(int, {1, 2, 3}); // error: the macro's arguments are "int", " {1", " 2", " 3}"
```

If you add the container type explicitly then it should work to just enclose the initializer_list in parens:
```suggestion
  NativeFunction(#NAME, 
std::vector<std::string> ALIASES, DataTypeVector{TYPE(), TYPE()}, TYPE(), \
```

Please add a function with at least two aliases (in a test would be acceptable) to demonstrate how the macro can be used with plural aliases.

Reply:
changed. Thanks for reviewing.

Reply:
To clarify: you don't need to wrap *all* initializer lists in parentheses, just those with more than a single element